app/api/relationships.v2/model/Relationship.ts

```diff
@@ -1,25 +1,92 @@
+/* eslint-disable max-params */
+/* eslint-disable max-classes-per-file */
 import { SelfReferenceError } from '../errors/relationshipErrors';
 
-export class Relationship {
+class Selection {
+  readonly page: number;
+
+  readonly top: number;
+
+  readonly left: number;
+
+  readonly height: number;
+
+  readonly width: number;
+
+  constructor(page: number, top: number, left: number, height: number, width: number) {
+    if (top < 0 || left < 0 || height <= 0 || width <= 0) {
+      throw new Error("Rectangle's top, left, height and width must be positive numbers");
+    }
+
+    if (page < 1) {
+      throw new Error('Page number must be positive');
+    }
+
+    this.page = page;
+    this.top = top;
+    this.left = left;
+    this.height = height;
+    this.width = width;
+  }
+}
+abstract class Pointer {
+  readonly entity: string;
+
+  constructor(entity: string) {
+    this.entity = entity;
+  }
+
+  abstract equals(pointer: Pointer): boolean;
+}
+
+class EntityPointer extends Pointer {
+  equals(pointer: EntityPointer): boolean {
+    return pointer instanceof EntityPointer && this.entity === pointer.entity;
+  }
+}
+
+class TextReferencePointer extends Pointer {
+  readonly file: string;
+
+  readonly selections: Selection[];
+
+  readonly text: string;
+
+  constructor(entity: string, file: string, selections: Selection[], text: string) {
+    if (!selections.length) {
+      throw new Error('Text references must have at least 1 selection');
+    }
+    super(entity);
+    this.file = file;
+    this.selections = selections;
+    this.text = text;
+  }
+
+  // eslint-disable-next-line class-methods-use-this
+  equals(_pointer: TextReferencePointer): boolean {
```

Comment:
Continuing the discussion: can we check how much this depends on user movement? It might be that the library that gives these selections calculates the bounding box on the selected text. If that is the case, then the same piece of text will always have the same selections, and it might be worth trying to equate the file and selections, even if we miss minute differences (like few pixels due an extra space at the end, etc.).